stac_fastapi/api/stac_fastapi/api/app.py

```diff
@@ -67,8 +68,8 @@ class StacApi:
 
     settings: ApiSettings = attr.ib()
     client: Union[AsyncBaseCoreClient, BaseCoreClient] = attr.ib()
-    extensions: List[ApiExtension] = attr.ib(default=attr.Factory(list))
-    exceptions: Dict[Type[Exception], int] = attr.ib(
+    extensions: list[ApiExtension] = attr.ib(default=attr.Factory(list))
+    exceptions: dict[Type[Exception], int] = attr.ib(
```

Comment:
FYI is using `dict` directly instead of `typing.Dict` is the reason to switch to python>=3.9, `typing.Dict` is just an alias of `dict` (https://github.com/python/cpython/blob/3.9/Lib/typing.py#L1743C15-L1743C19) so it's fine to use `typing.Dict` IMO 

https://stackoverflow.com/questions/37087457/difference-between-defining-typing-dict-and-dict

Reply:
same goes for List

Reply:
@vincentsarago I found that during the `create_request_model` in models.py. The `field_info.annotation` would leave out the `Optional` part of the type. The only solution I found was to switch from typing List and Dict to the generic list and dict. There maybe a better solution to this?